yb-voyager/src/queryparser/unsupported_constructs.go

```diff
@@ -0,0 +1,121 @@
+package queryparser
+
+import (
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/protobuf/reflect/protoreflect"
+)
+
+const (
+	ADVISORY_LOCKS = "Advisory Locks"
+	SYSTEM_COLUMNS = "System Columns"
+	XML_FUNCTIONS  = "XML Functions"
+)
+
+// To Add a new unsupported query construct implement this interface
+type UnsupportedConstructDetector interface {
+	Detect(msg protoreflect.Message) ([]string, error)
+}
+
+type FuncCallDetector struct {
+	// right now it covers Advisory Locks and XML functions
+	unsupportedFuncs map[string]string
+}
+
+func NewFuncCallDetector() *FuncCallDetector {
+	fcd := &FuncCallDetector{
+		unsupportedFuncs: make(map[string]string),
+	}
+
+	for _, fname := range unsupportedAdvLockFuncs {
+		fcd.unsupportedFuncs[fname] = ADVISORY_LOCKS
+	}
+	for _, fname := range xmlFunctions {
+		fcd.unsupportedFuncs[fname] = XML_FUNCTIONS
+	}
+	log.Debug("initialized FuncCallDetector with unsupported functions")
+	return fcd
+}
+
+// Detect checks if a FuncCall node uses an unsupported function.
+func (d *FuncCallDetector) Detect(msg protoreflect.Message) ([]string, error) {
+	if getMsgFullName(msg) != PG_QUERY_FUNCCALL_NODE {
+		return nil, nil
+	}
+
+	funcName := getFuncNameFromFuncCall(msg)
+	log.Debugf("fetched function name from %s node: %q", PG_QUERY_FUNCCALL_NODE, funcName)
+	if constructType, ok := d.unsupportedFuncs[funcName]; ok {
```

Comment:
nit for readability : 
`if constructType, isUnsupported := d.unsupportedFuncs[funcName]; isUnsupported {`